cmd/podman/kube/play.go

```diff
@@ -278,16 +286,58 @@ func play(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	if err := kubeplay(reader); err != nil {
-		// teardown any containers, pods, and volumes that might have created before we hit the error
-		teardownReader, trErr := readerFromArg(args[0])
-		if trErr != nil {
-			return trErr
+	// Create a channel to catch an interrupt or SIGTERM signal
+	ch := make(chan os.Signal, 1)
+	var teardownReader *bytes.Reader
+	if playOptions.Wait {
+		// Stop the the shutdown signal handler so we can actually clean up after a SIGTERM or interrupt
+		if err := shutdown.Stop(); err != nil {
+			return err
+		}
+		signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
+		playOptions.ServiceContainer = true
+
+		// Read the kube yaml file again so that a reader can be passed down to the teardown function
+		teardownReader, err = readerFromArg(args[0])
```

Comment:
This can be passed down to the routine that actually needs the data.  Also erases the need for the variable.

Reply:
I am using this in line 342 as well, for the case when we have to clean up after the pods exit. This happens only when --wait=true, so reading the kube yaml here and saving it to be used for clean up either when it is killed or when the pods exit. This way I don't have to repeat the `readerFromArg` code twice.